### ISSUE TITLE
Should GitCommitsCheck flag this as an error?

### ISSUE THREAD

**Issue by ezzieyguywuf** [pkgcore/pkgcheck#290]
I have a branch with a commit message that reads `dev-haskell yaml-light 0.1.4: rerun hackport, update eapi`.

I ran `pkgcheck scan --commits upstream/master --checks +GitCommitsCheck`, but it did not flag this as a commit message.

Am I doing something wrong? (don't say "yea, you're not using `repoman commit` :-P)

**Comment by radhermit** [comment]
> I have a branch with a commit message that reads `dev-haskell yaml-light 0.1.4: rerun hackport, update eapi`.
> 
> I ran `pkgcheck scan --commits upstream/master --checks +GitCommitsCheck`, but it did not flag this as a commit message.

`pkgcheck scan --commits` is set up to use 'origin' and 'master' branches by default (using origin..HEAD as the default ref for diff generation). If you want to target the 'upstream' branch instead you'd probably do something like `pkgcheck scan --commits upstream..HEAD`, adding `--checks +GitCommitsCheck` is unnecessary as git checks are enabled automatically when using the `--commits` option.

I probably could making it easier to debug by outputting the related `git diff-tree` output that is used for restriction generation when `--debug` is enabled.

> Am I doing something wrong? (don't say "yea, you're not using `repoman commit` :-P)

I haven't used repoman in years so I'd never say that. :wink:

**Comment by ezzieyguywuf** [comment]
>  If you want to target the 'upstream' branch instead you'd probably do something like `pkgcheck scan --commits upstream..HEAD`

Tried this, got `pkgchck scan: error: failed running git: fatal: ambiguous argument 'upstream..HEAD'`

>  adding `--checks +GitCommitsCheck` is unnecessary as git checks are enabled automatically when using the `--commits` option.

I don't find this to be the case.

```sh
pkgcheck --scan --commits
pkgcheck --scan --commits --checks +GitCommitsCheck
WARNING pkgcheck: running gentoo specific check: GitCommitsCheck

commit
    BadCommitSummary:....
```

**Comment by radhermit** [comment]
> > If you want to target the 'upstream' branch instead you'd probably do something like `pkgcheck scan --commits upstream..HEAD`
> 
> Tried this, got `pkgchck scan: error: failed running git: fatal: ambiguous argument 'upstream..HEAD'`

I'll try to reproduce this later and see what I can do to fix it and make it easier to debug.
 
> > adding `--checks +GitCommitsCheck` is unnecessary as git checks are enabled automatically when using the `--commits` option.
> 
> I don't find this to be the case.
> 
> ```shell
> pkgcheck --scan --commits
> pkgcheck --scan --commits --checks +GitCommitsCheck
> WARNING pkgcheck: running gentoo specific check: GitCommitsCheck
> 
> commit
>     BadCommitSummary:....
> ```

Ah sorry you're correct, I forgot GitCommitsCheck currently defaults to only running for the gentoo repo.
